apps/server/src/modules/task-card/uc/task-card.uc.ts

```diff
@@ -135,14 +132,10 @@ export class TaskCardUc {
 			name: params.title,
 			courseId: params.courseId,
 			dueDate: params.dueDate,
-			availableDate: new Date(),
+			availableDate: params.visibleAtDate,
```

Comment:
see above

Reply:
if it's not set the update doesn't work correctly. There has to be the availableDate-property declared and set to undefined. Otherwise it will not be deleted.

Reply:
if it needs to be "at least undefined", why not rewrite the `updateTask` for example to the following to make the needed logic reflected in code:
```
	private async updateTask(userId: EntityId, id: EntityId, params: ITaskCardCRUD) {
		const taskParams: ITaskUpdate = {
			name: params.title,
			courseId: params.courseId,
			dueDate: params.dueDate,
			availableDate: undefined,
		};

		if (params.visibleAtDate) {
			taskParams.availableDate = params.visibleAtDate;
		}

		const taskWithStatusVo = await this.taskService.update(userId, id, taskParams);

		return taskWithStatusVo;
	}
```